webui/react/src/pages/TrialDetails/F_TrialDetailsOverview.tsx

```diff
@@ -56,19 +56,30 @@ const TrialDetailsOverview: React.FC<Props> = ({ experiment, trial }: Props) =>
     title: `Best checkpoint for Trial ${trial?.id}`,
   });
 
+  const { metrics, data } = useTrialMetrics(trial);
+
   const checkpointsDict = useMemo<CheckpointsDict>(() => {
-    return checkpoint && xAxis === XAxisDomain.Batches
-      ? {
-          [checkpoint?.totalBatches]: checkpoint,
+    const timeHelpers: Record<XAxisVal, CheckpointWorkloadExtended> = {};
+    if (data && xAxis === XAxisDomain.Time && checkpoint?.totalBatches) {
+      Object.values(data).forEach((metric) => {
+        const matchIndex = metric.data[XAxisDomain.Batches]?.findIndex(
+          (pt) => pt[0] === checkpoint.totalBatches,
+        );
+        if (matchIndex !== undefined && matchIndex >= 0) {
```

Comment:
Just wanted to note that it seems like this logic will need to be updated when `epochs` are introduced but happy to defer that for now.

Reply:
I agree; my idea while going over these is that LineChart will need to put checkpoints in a new series